tool/tctl/common/bots_command.go

```diff
@@ -494,3 +643,52 @@ func splitRoles(flag string) []string {
 	}
 	return roles
 }
+
+// setsEqual determines if two sets contain the same keys.
+func setsEqual[T comparable](a map[T]struct{}, b map[T]struct{}) bool {
```

Comment:
This set implementation is pretty clean and generic, can we move it to some shared lib? I'm constantly reimplementing sets with empty maps and would benefit from this code.

Reply:
I've spun it out into `github.com/gravitational/teleport/lib/utils/set` to make it more reusable (and adjusted the API to make it a bit more sane than flinging `map[T]struct{}{}` around)